ansible/roles/debops.etherpad/defaults/main.yml

```diff
@@ -465,6 +483,41 @@ etherpad__mariadb__dependent_databases:
   - database: '{{ etherpad_database_name }}'
 
                                                                    # ]]]
+# .. envvar:: etherpad__postgresql__dependent_roles [[[
+#
+# User configuration for the debops.postgresql_ Ansible role.
+etherpad__postgresql__dependent_roles:
+
+  - name: '{{ etherpad_database_user }}'
+    password: '{{ etherpad_database_password }}'
+    flags: [ 'NOSUPERUSER', 'NOCREATEDB', 'LOGIN' ]
+
+                                                                   # ]]]
+# .. envvar:: etherpad__postgresql__dependent_groups [[[
+#
+# Group configuration for the debops.postgresql_ Ansible role.
+etherpad__postgresql__dependent_groups:
+
+                                                                   # ]]]
```

Comment:
The common way of setting up PostgreSQL databases and users in DebOps is to create two roles:

- role for the database that cannot directly login, for example `etherpad_lite_production`
- role for the application user that can login, has a password and has access to the database role, for example `etherpad_lite`
That way, permissions in the database are somewhat manageable. you can check the `debops.gitlab` or `debops.netbox` roles for examples.

You used only one role in your PR, and I guess that's fine as well. However if you go that route, you shouldn't leave empty variables like this one and `etherpad__postgresql__dependent_pgpass` below. Either remove them, and their corresponding entries in the playbooks, or switch to the separated PostgreSQL roles.

Reply:
Right now, the database name and user are based off of `etherpad_system_name` as are many other variables. The default value of `etherpad_system_name` is `etherpad-lite` but you suggest `etherpad_lite_production` for the database role name--looking at `debops.gitlab` and `debops.netbox` the database name and role name are ideally the same. How should this be handled here? Should the `etherpad_database_name` change or the `etherpad_system_name` change? Either way, I believe the system user and database user name need to be the same for a unix socket connection. I just don't know which way to go as this could change a number of other things.

Reply:
@drybjed any thoughts on this?

Reply:
Ah, sorry, must have missed it. The way `debops.postgresql_server` configures `pg_hba.conf`, you can create two roles:

- one role as a "database holder", `etherpad_lite_production`

- another role as an "application client", `etherpad_lite`

Now, when you add the `etherpad_lite` role to the `etherpad_lite_production` role inside PostgreSQL, the `etherpad_lite` will be able to login and access the `etherpad_lite_production` database. For this to work, application needs to explicitly request that database, otherwise the connection will be refused because there's no `etherpad_lite` database.

I'm not sure if that's the case with Etherpad. If it is, ie. the role name and database name needs to be the same, then I guess that we can stay with the current configuration; just remove the empty default variables.

Reply:
That is the case with `debops.etherpad`. The role currently has the postgres user and database with the same name: `'etherpad-lite'` (with a hyphen as opposed to an underscore). These are both based off of `etherpad_system_name` (set to `'etherpad-lite'`) as are a number of other variables in the role.

I had intended on changing the database name to append `'_production'` and changing both the db name and db user name to use underscores (`'etherpad_lite'` as opposed to `'etherpad-lite'` to be clear) to be more like the `netbox` and `gitlab` roles but then I realized I should probably also change `etherpad_system_name` (to `'etherpad_lite'`) but that would change a lot as so many other things build off of this variable in the role. I believe such a change however would be disruptive to anyone who may already be using this role.

I'm happy to make the changes but what do you prefer in this case?

Reply:
In that case let's stay with current setup, `etherpad-lite`. I'm trying to preserve current DebOps roles as-is, fixing any bugs so that the existing setup can be released as `stable-1.0` branch. More disruptive changes can be done after that, and if somebody will want to stick to the current setup, they will be able to by selecting a specific branch.